mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/persistence/auth/firebase/FirebaseAuth.kt

```diff
@@ -38,28 +38,38 @@ class FirebaseAuth(private val actual: ActualFirebaseAuth) : Auth {
    */
   private suspend fun authenticate(
       block: suspend () -> AuthResult?,
+      type: String,
   ): AuthenticationResult =
       try {
         val result = block()
         val user = result?.user?.let(::FirebaseUser)
         AuthenticationResult.Success(user)
       } catch (other: Throwable) {
-        AuthenticationResult.FailureInternal
+        when (other) {
+          is FirebaseAuthWeakPasswordException -> AuthenticationResult.FailureBadPassword
+          is FirebaseAuthInvalidCredentialsException ->
+              when (type) {
+                "SignIn" -> AuthenticationResult.FailureIncorrectPassword
+                "SignUp" -> AuthenticationResult.FailureIncorrectEmailFormat
```

Comment:
I don't really like these magic values...(it's error prone) Maybe transform this into a enum

Reply:
Fixed in a362d0a